.github/dependabot.yml

```
@@ -11,3 +11,9 @@ updates:
       interval: daily
     labels:
       - "topic: infrastructure"
+  - package-ecosystem: gomod
+    directory: /
+    schedule:
+      interval: monthly
```

```suggestion
      interval: daily
```

The reason I prefer the daily interval is this causes the PRs for a given bump to be received in all repositories that have that dependency on the same day.

I have found that it is efficient to do reviews of these PRs in quick succession because I can evaluate the changeset once and have that information fresh in my mind for all the reviews. In the case where there is some complexity involved in the bump, it is more efficient to make the necessary adjustments to all affected projects in quick succession.

With the alternative approach of longer intervals between applying a given bump to each project, it is necessary to duplicate the effort of evaluating the changeset and determining how to overcome any complications anew for each project because my memory of the exact details of the last time I performed that bump has grown fuzzy with time.

---

I've put monthly after a brief discussion in a daily standup. @cmaglie raised the concern that we would end up being spammed by each bump, and we might not keep up with the testing of those bumps.
So we agreed to enable this by starting with a monthly interval and if we're happy to lower it to weekly/daily.

@cmaglie @umbynos @MatteoPologruto After @per1234 insight we still want to try starting with interval monthly?

Personally, I'm okay with setting it to daily/weekly. We're not obliged to merge dependabot PR instantly, so if a new version of a package is being released the day after, dependabot is smart enough to just update the already opened PR without causing too much noise.

---

> Personally, I'm okay with setting it to daily/weekly. We're not obliged to merge dependabot PR instantly, so if a new version of a package is being released the day after, if I'm not wrong dependabot is smart enough to just update the already opened PR without causing too much noise.

We have the `daily` dependabot automation in place [here](https://github.com/arduino/arduino-create-agent/blob/main/.github/dependabot.yml) and it's not too spammy because we have the ` open-pull-requests-limit: 10` setting and it correctly closes and reopens a PR to the new version if a new one comes out (example: https://github.com/arduino/arduino-create-agent/pull/829)

---

My review was only a suggestion based on my own experience. The preferences of the primary maintainers of the project are paramount.

> because we have the  open-pull-requests-limit: 10 setting

And that custom configuration is even higher than the default value of 5 that would be in effect with the current proposal from this PR:

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#open-pull-requests-limit

I do personally think I would prefer a higher number because sometimes an accumulation of more challenging bumps builds up and that can prevent Dependabot from submitting additional PRs that would be welcome and easily merged. I didn't notice the default was so low until I had already set up Dependabot in the repos I maintain.

---

Let's try `daily` and see how it goes.